### DISCUSSION TITLE
Add attribute combinations to virtual products

### DISCUSSION THREAD

**Discussion by dali-rajab** [PrestaShop/PrestaShop#32626] [Ideas]
### The problem

Hello, I was wondering why dematerialized products (virtual products) in Prestashop cannot have variations (attribute combinations). I have noticed that there are modules available to enable this functionality, but why is it not integrated as a native feature?

### The solution

I'm suggesting refactoring some code to enable that feature

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by eternoendless** [comment]
I think it's a good idea, but to the best of my understanding, there are implementation constraints that aren't easy to circumvent to allow this.
